pkg/rbac/sharedData.go

```diff
@@ -210,3 +221,137 @@ func (shared *SharedData) GetManagedClusters(cache *Cache, ctx context.Context)
 	return shared.mcErr
 
 }
+
+// Returns a map of managed clusters for which the search add-on has been disabled.
+func (cache *Cache) GetDisabledClusters(ctx context.Context) (*map[string]struct{}, error) {
+	userData, userDataErr := cache.GetUserData(ctx)
+	if userDataErr != nil {
+		return nil, userDataErr
+	}
+	// lock to prevent the query from running repeatedly
+	cache.shared.dcLock.Lock()
+	defer cache.shared.dcLock.Unlock()
+
+	if !cache.sharedCacheDisabledClustersValid() {
+		klog.V(5).Info("DisabledClusters cache empty or expired. Querying database.")
+		// - running query to get search addon disabled clusters")
+		//run query and get disabled clusters
+		if disabledClustersFromQuery, err := cache.findSrchAddonDisabledClusters(ctx); err != nil {
+			klog.Error("Error retrieving Search Addon disabled clusters: ", err)
+			cache.setDisabledClusters(map[string]struct{}{}, err)
+			return nil, err
+		} else {
+			cache.setDisabledClusters(*disabledClustersFromQuery, nil)
+		}
+	}
+
+	//check if user has access to disabled clusters
+	userHasAccess, userAccessClusters := userHasAccessToDisabledClusters(&cache.shared.disabledClusters,
+		userData.ManagedClusters)
+	if userHasAccess {
+		klog.V(5).Info("user has access to Search Addon disabled clusters ")
+		return &userAccessClusters, cache.shared.dcErr
+
+	} else {
+		klog.V(5).Info("user does not have access to Search Addon disabled clusters ")
+		return &map[string]struct{}{}, nil
+	}
+
+}
+
+func userHasAccessToDisabledClusters(disabledClusters *map[string]struct{}, userClusters []string) (bool,
```

Comment:
- Can simplify removing the returned bool. It's the same as `len(userAccessDisabledClusters) > 0`
- Maybe rename to something like `disabledClustersForUser()